### PR TITLE
Fixes testing of C# kernels

### PR DESCRIPTION
**Pull request by DonJayamanne**
Enabled to get logs

### CODE REVIEW

File: src/client/common/process/logger.ts [hunk 1]
```diff
@@ -38,7 +38,7 @@ export class ProcessLogger implements IProcessLogger {
         }
 
         info.forEach((line) => {
-            traceInfo(line);
+            traceInfo(`Process Execution: ${line}`);
```
Comment (DonJayamanne) [inline src/client/common/process/logger.ts:41]: It was difficult to tell where the message `cwd: <path` was coming from.
This helps us on CI figure out the source of such messages.

File: src/client/datascience/jupyter/kernels/cellExecution.ts [hunk 2]
```diff
@@ -677,9 +677,11 @@ export class CellExecution {
                 }
                 // tslint:disable-next-line:restrict-plus-operands
                 existing.data['text/plain'] = formatStreamText(concatMultilineString(`${existingOutput}${newContent}`));
+                traceInfoIf(!!process.env.VSC_JUPYTER_LOG_KERNEL_OUTPUT, `Append Output '${existing.data['text/plain']}'`);
                 edit.replaceCellOutput(this.cell.index, [...exitingCellOutput]); // This is necessary to get VS code to update (for now)
             } else {
                 const originalText = formatStreamText(concatMultilineString(msg.content.text));
+                traceInfoIf(!!process.env.VSC_JUPYTER_LOG_KERNEL_OUTPUT, `Insert Output '${originalText}'`);
```
Comment (DonJayamanne) [inline src/client/datascience/jupyter/kernels/cellExecution.ts:687]: This helped figure out why the tests were failing.
Leaving for now, as I have another test thats flaky (stderr & stdout test).
Also this stuff never gets logged in production & this variable is only initialized in a specific test, hence no mark for now.

File: src/client/datascience/jupyter/kernels/kernelSelector.ts [hunk 3]
```diff
@@ -535,6 +535,10 @@ export class KernelSelector implements IKernelSelectionUsage {
 
         // First use our kernel finder to locate a kernelspec on disk
         const kernelSpec = await this.kernelFinder.findKernelSpec(resource, notebookMetadata, cancelToken);
+        traceInfoIf(
+            !!process.env.VSC_JUPYTER_FORCE_LOGGING,
+            `Kernel spec found ${JSON.stringify(kernelSpec)}, metadata ${JSON.stringify(notebookMetadata || '')}`
```
Comment (DonJayamanne) [inline src/client/datascience/jupyter/kernels/kernelSelector.ts:540]: Useful on CI.
Found that C# kernels do not work when no jupyter and tests were failing, this helped.
Leaving for now, as its useful (for CI).

File: src/test/datascience/notebook/helper.ts [hunk 4]
```diff
@@ -213,11 +218,13 @@ export async function waitForKernelToGetAutoSelected(expectedLanguage?: string)
...
             return false;
         }
-        if (!expectedLanguage) {
-            kernelInfo = '<No specific kernel expected>';
-            return true;
-        }
         if (isJupyterKernel(vscodeNotebook.activeNotebookEditor.kernel)) {
+            if (!expectedLanguage) {
+                kernelInfo = `<No specific kernel expected> ${JSON.stringify(
+                    vscodeNotebook.activeNotebookEditor.kernel.selection
```
Comment (DonJayamanne) [inline src/test/datascience/notebook/helper.ts:224]: Moved the code so we can see what kernel is selected on CI (for logging puproses).
FYI - this is test code.

File: src/client/datascience/jupyter/kernels/cellExecution.ts [hunk 5]
```diff
@@ -172,7 +172,7 @@ export class CellExecution {
         }
         traceCellMessage(this.cell, 'Start execution');
         traceInfoIf(
-            !!process.env.VSC_JUPYTER_LOG_KERNEL_OUTPUT,
+            !!process.env.VSC_JUPYTER_FORCE_LOGGING,
```
Comment (DonJayamanne) [inline src/client/datascience/jupyter/kernels/cellExecution.ts:175]: Was useful, hence logging this always on CI.

File: src/test/datascience/notebook/helper.ts [hunk 6]
```diff
@@ -552,24 +563,17 @@ export async function hijackPrompt(
...
             displayed.resolve(true);
             if (buttonToClick) {
                 return clickButton.promise;
             }
-        } else {
-            console.info(`Message not found, looking for message ${JSON.stringify(message)}`);
-            console.info(`Exact match ${'exactMatch' in message ? 'true' : 'false'}`);
-            console.info(`Exact match ${'exactMatch' in message && msg.trim() === message.exactMatch.trim()}`);
-            console.info(`EndsWith ${'endsWith' in message ? 'true' : 'false'}`);
-            console.info(`EndsWith ${'endsWith' in message && msg.endsWith(message.endsWith)}`);
```
Comment (DonJayamanne) [inline src/test/datascience/notebook/helper.ts:572]: Problem fix, hence no longer required.

File: src/test/datascience/notebook/nonPythonKernels.vscode.test.ts [hunk 7]
```diff
@@ -173,7 +169,7 @@ suite('DataScience - VSCode Notebook - Kernels (non-python-kernel) (slow)', () =
         this.timeout(30_000); // Can be slow to start Julia kernel on CI.
         await openNotebook(api.serviceContainer, testJuliaNb.fsPath);
         await insertCodeCell('123456', { language: 'julia', index: 0 });
-        await waitForKernelToGetAutoSelected();
+        await waitForKernelToGetAutoSelected('julia');
```
Comment (DonJayamanne) [inline src/test/datascience/notebook/nonPythonKernels.vscode.test.ts:172]: Found that C# was defaulting to Julia kernel when Jupyter wasn't available.
This way when tests fail, we know exactly why its failing. I.e. fail early when the wrong kernel is selected.

**Review by rchiodo — APPROVED** [review]

**Review by DavidKutu — APPROVED** [review]

**Pull request merged**